README.md
# IPventur.sh
IP collector script for beginners.
This little helping tool, collects IP addresses and open ports with nmap and fping (need to be installed).
To start the tool and invest MAC addresses as well:

chmod +x IPventur-eng-basic.sh (or chmod +x IPventur-ger-basic.sh)

sudo ./IPventur-eng-basic.sh 192.168.1.0/24 <=network example, always use netmask (German version IPventur-ger-basic.sh)

The script will check first, if nmap and fping available, and when yes, it will starts with the investigation.

Feel free to change the nmap parameters with your own wishes!

Happy collecting!

Mr.Energy

P.S. if you would like to get more details or use scripts, check out my IPventur-Pro version!


CHANGELOG.TXT
Version 0.5 origin start script IPventur-eng.sh amd IPventur-ger.sh with standard output and scan and MAC adress

Version 0.6 IPventur-eng-06.sh/IPventur-ger-06.sh - give more comments in the script and extend nmap
            parameters to -A -T5 and add Vendor ID's to the MAC address output

Version 0.7 - change it from sh to bash script
            - added date to the output file
            - rename script to IPventur-eng-basic.sh (or IPventur-ger-basic.sh)
            - minor style changes
            

Note: if you would like to get more scan details or different output formats, use IPventur-Pro!


IPventur-eng.sh
#! /bin/sh
# IPventur.sh
# need root rights, fping and nmap
# updated: 23.05.2020 mr.energy origin: Linux-User
# Version 0.5
#
# Start without parameter - exit
clear
if [ -z $1 ]; then
	echo "NETWORK/CIDR (e.g. 192.168.0.0/24) is needed!"
	exit
fi
clear
echo "-------------------------------------------------------"
# check if fping and nmap installed
echo "-------------------------------------------------------"
echo
command -v fping >/dev/null 2>&1 || { echo >&2 "I require fping but it's not installed.  Please install."; exit 1; }
echo Program fping is installed!
echo
command -v nmap >/dev/null 2>&1 || { echo >&2 "I require nmap but it's not installed.  Please install."; exit 1; }
echo Program nmap is installed!
echo
echo
echo "Overview active network systems:"
echo "(to get MAC addresses, start the search network as root/sudo)"
echo "--------------------------------------------------------------"
echo
date2=$(date +%d.%m.%Y-%H:%M:%S)
echo Start: $date2  Net/IP = $1
echo
net2=$(echo $1 | cut -d "/" -f 1)
netO=$net2
# Begin creation output file
echo "Network items $datum / $1" > lanlist-$netO.txt
echo "-------------------------------------------------------" >> lanlist-$netO.txt
echo "-------------------------------------------------------"

# Scan network and log to the output file
for k in $(fping -aq -g $1); do
	echo "scanning...: $k"
	echo "Online: $k" >> lanlist-$netO.txt
	nmap -n -sP $k | awk '/Nmap scan report for/{printf $5;}/MAC Address:/{print " => "$3;}' | sort >> lanlist-$netO.txt
	nmap $k | grep -B1 open >> lanlist-$netO.txt
	echo "---------------------------------------------------" >> lanlist-$netO.txt
done
echo "-------------------------------------------------------"
echo "              E N D" >> lanlist-$netO.txt
echo
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $date2 >> lanlist-$netO.txt
echo "-------------------------------------------------------" >> lanlist-$netO.txt
echo End: $date2
echo 
# Display output file
less lanlist-$netO.txt


IPventur-ger.sh
#! /bin/sh
# IPventur.sh
# need root rights, fping und nmap
# updated: 23.05.2020 nwoeske origin: Linux-User
# Version: 0.5
#
# Bei Aufruf ohne Parameter abbrechen
clear
if [ -z $1 ]; then
	echo "NETZADRESSE/CIDR muss beim Aufruf angegeben werden!"
	exit
fi
clear
echo "-------------------------------------------------------"
# Check for fping and nmap
echo "-------------------------------------------------------"
echo
command -v fping >/dev/null 2>&1 || { echo >&2 "fping ist nicht installiert!  Bitte installieren."; exit 1; }
echo "Programm fping ist vorhanden!"
echo
command -v nmap >/dev/null 2>&1 || { echo >&2 "nmap ist nicht installiert!  Bitte installieren."; exit 1; }
echo "Programm nmap ist vorhanden!"
echo 
echo
echo "Übersicht aktiver Netzwerkteilnehmer:"
echo "(für MAC Adressen: als root/sudo im selben Netzwerk starten)"
echo "------------------------------------------------------------"
echo
echo
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $datum  Netz/IP = $1
netz2=$(echo $1 | cut -d "/" -f 1)
netz=$netz2
echo
# Beginn Ergebnisdatei
echo "Netzwerkbestand $datum / $1" > lanliste-$netz.txt
echo "-------------------------------------------------------" >> lanliste-$netz.txt
echo "-------------------------------------------------------"

# Scannen des Netztes und Ablage in Ergebnisdatei
for k in $(fping -aq -g $1); do
	echo "wird untersucht: $k"
	echo "Aktiv: $k" >> lanliste-$netz.txt
	nmap -n -sP $k | awk '/Nmap scan report for/{printf $5;}/MAC Address:/{print " => "$3;}' | sort >> lanliste-$netz.txt
	nmap $k | grep -B1 open >> lanliste-$netz.txt
	echo "---------------------------------------------------" >> lanliste-$netz.txt
done
echo "-------------------------------------------------------"
echo "              E N D E" >> lanliste-$netz.txt
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $datum >> lanliste-$netz.txt
echo "-------------------------------------------------------" >> lanliste-$netz.txt
echo Ende: $datum
echo
# Anzeige Ergebnisdatei
less lanliste-$netz.txt


IPventur-eng-06.sh
#! /bin/sh
# IPventur.sh
# need root rights, fping and nmap
# updated: 08.06.2020 MrEnergy64 origin: Linux-User
# Version: 0.6
#
# Start without parameter - exit
clear
if [ -z $1 ]; then
	echo 
	echo "NETWORK/CIDR (e.g. 192.168.0.0/24) is needed!"
	echo "(Example: sudo ./IPventur-eng-06.sh 10.0.0.0/23 [Enter])"
	exit
fi
clear
echo "-------------------------------------------------------"
# check if fping and nmap installed
echo "-------------------------------------------------------"
echo
command -v fping >/dev/null 2>&1 || { echo >&2 "I require fping but it's not installed.  Please install."; exit 1; }
echo Program fping is installed!
echo
command -v nmap >/dev/null 2>&1 || { echo >&2 "I require nmap but it's not installed.  Please install."; exit 1; }
echo Program nmap is installed!
echo
echo
echo "Overview active network systems:"
echo "(to get MAC addresses, start the search network as root/sudo)"
echo "--------------------------------------------------------------"
echo
date2=$(date +%d.%m.%Y-%H:%M:%S)
echo Start: $date2  Net/IP = $1
echo
net2=$(echo $1 | cut -d "/" -f 1)
netO=$net2
# Begin creation output file
echo "Network items $datum / $1" > lanlist-$netO.txt
echo "-------------------------------------------------------" >> lanlist-$netO.txt
echo "-------------------------------------------------------"

# Scan network and log to the output file, fping checks online IP's only
for k in $(fping -aq -g $1); do
	echo "scanning...: $k"
	echo "Online: $k" >> lanlist-$netO.txt
# -n Never do DNS resolution resolve, -sP check if IP up and give MAC address with Vendor
	nmap -n -sP $k | awk '/Nmap scan report for/{printf $5;}/MAC Address:/{print " => "$3" "$4" "$5;}' | sort >> lanlist-$netO.txt
# -A Enable OS detection, version detection, script scanning, and tracerout, -T<0-5>: Set timing template (higher is faster) 
	nmap -A -T5 $k | grep -B1 open >> lanlist-$netO.txt
	echo "---------------------------------------------------" >> lanlist-$netO.txt
done
echo "-------------------------------------------------------"
echo "              E N D" >> lanlist-$netO.txt
echo
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $date2 >> lanlist-$netO.txt
echo "-------------------------------------------------------" >> lanlist-$netO.txt
echo End: $date2
echo 
# Display output file
less lanlist-$netO.txt


IPventur-ger-06.sh
#! /bin/sh
# IPventur.sh
# benötigt root rechte, sowie fping und nmap
# updated: 08.06.2020 MrEnergy64 origin: Linux-User
# Version: 0.6
#
# Bei Aufruf ohne Parameter abbrechen
clear
if [ -z $1 ]; then
	echo
	echo "NETZADRESSE/Subnetmask muss beim Aufruf angegeben werden!"
	echo "(Beispiel: sudo ./IPventur-ger-06.sh 10.0.0.0/23 [Enter])"
	exit
fi
clear
echo "-------------------------------------------------------"
# Check for fping and nmap
echo "-------------------------------------------------------"
echo
command -v fping >/dev/null 2>&1 || { echo >&2 "fping ist nicht installiert!  Bitte installieren."; exit 1; }
echo "Programm fping ist vorhanden!"
echo
command -v nmap >/dev/null 2>&1 || { echo >&2 "nmap ist nicht installiert!  Bitte installieren."; exit 1; }
echo "Programm nmap ist vorhanden!"
echo 
echo
echo "Übersicht aktiver Netzwerkteilnehmer:"
echo "(für MAC Adressen: als root/sudo im selben Netzwerk starten)"
echo "------------------------------------------------------------"
echo
echo
# aktuelles Startdatum und Uhrzeit wird der Ausgangsdatei hinzugefügt
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $datum  Netz/IP = $1
netz2=$(echo $1 | cut -d "/" -f 1)
netz=$netz2
echo
# Beginn Ergebnisdatei
echo "Netzwerkbestand $datum / $1" > lanliste-$netz.txt
echo "-------------------------------------------------------" >> lanliste-$netz.txt
echo "-------------------------------------------------------"

# Scannen des Netztes und Ablage in Ergebnisdatei, fping stellt fest welche IP Online ist
for k in $(fping -aq -g $1); do
	echo "wird untersucht: $k"
	echo "Aktiv: $k" >> lanliste-$netz.txt
# -n startet kein IP DNS Namen check (nur IP Adresse), -sP zeigt IP Adresse mit MAC Adresse und Hersteller ID
	nmap -n -sP $k | awk '/Nmap scan report for/{printf $5;}/MAC Address:/{print " => "$3" "$4" "$5;}' | sort >> lanliste-$netz.txt
# -A zeigt mehr Informationen wie OS Version, Traceroute, welche Scripts laufen etc.
	nmap -A -T5 $k | grep -B1 open >> lanliste-$netz.txt
	echo "---------------------------------------------------" >> lanliste-$netz.txt
echo >> lanliste-$netz.txt
done
echo "-------------------------------------------------------"
echo "              E N D E" >> lanliste-$netz.txt
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $datum >> lanliste-$netz.txt
echo "-------------------------------------------------------" >> lanliste-$netz.txt
echo Ende: $datum
echo
# Anzeige Ergebnisdatei
less lanliste-$netz.txt


IPventur-ger-basic.sh
#! /bin/bash
# IPventur-basic.sh
# benötigt root rechte, sowie fping und nmap
# updated: 11.06.2020 MrEnergy64 origin: Linux-User
# Version: 0.7
#
# Bei Aufruf ohne Parameter abbrechen
clear
if [ -z $1 ]; then
	echo ""
	echo "******************************"
	echo "* IPventur-Deutsch Basic 0.7 *"
	echo "******************************"
	echo ""
	echo "NETZADRESSE/Subnetmask muss beim Aufruf angegeben werden!"
	echo "(Beispiel: sudo ./IPventur-ger-06.sh 10.0.0.0/23 oder 192.168.1.1/32 [Enter])"
	exit
fi
clear
echo "-------------------------------------------------------"
# Check for fping and nmap
echo "-------------------------------------------------------"
	echo ""
	echo "******************************"
	echo "* IPventur-Deutsch Basic 0.7 *"
	echo "******************************"
	echo ""
command -v fping >/dev/null 2>&1 || { echo >&2 "fping ist nicht installiert!  Bitte installieren."; exit 1; }
echo "Programm fping ist vorhanden!"
echo
command -v nmap >/dev/null 2>&1 || { echo >&2 "nmap ist nicht installiert!  Bitte installieren."; exit 1; }
echo "Programm nmap ist vorhanden!"
echo 
echo
echo "Übersicht aktiver Netzwerkteilnehmer:"
echo "(für MAC Adressen: als root/sudo im selben Netzwerk starten)"
echo "------------------------------------------------------------"
echo
# aktuelles Startdatum und Uhrzeit wird der Ausgangsdatei hinzugefügt
datum=$(date +%d.%m.%Y-%H:%M:%S)
datum2=$(date +%d%m%Y)
echo Start: $datum
echo Kommando: nmap -A -T5 $1
echo
netz2=$(echo $1 | cut -d "/" -f 1)
netz=$netz2
# Beginn Ergebnisdatei
echo "-------------------------------------------------------" > lanliste-$netz-$datum2.txt
echo "Netzwerkbestand $datum / nmap -A -T5 $1" >> lanliste-$netz-$datum2.txt
echo "-------------------------------------------------------"
# Scannen des Netztes und Ablage in Ergebnisdatei, fping stellt fest welche IP Online ist
for k in $(fping -aq -g $1); do
	echo "wird untersucht: $k"
	echo "Aktiv: $k" >> lanliste-$netz-$datum2.txt
# -n startet kein IP DNS Namen check (nur IP Adresse), -sP zeigt IP Adresse mit MAC Adresse und Hersteller ID
	nmap -n -sP $k | awk '/Nmap scan report for/{printf $5;}/MAC Address:/{print " => "$3" "$4" "$5;}' | sort >> lanliste-$netz-$datum2.txt
# -A zeigt mehr Informationen wie OS Version, Traceroute, welche Scripts laufen etc.
	nmap -A -T5 $k | grep -B1 open >> lanliste-$netz-$datum2.txt
	echo "---------------------------------------------------" >> lanliste-$netz-$datum2.txt
echo >> lanliste-$netz-$datum2.txt
done
echo "-------------------------------------------------------"
echo "                      E N D E" >> lanliste-$netz-$datum2.txt
echo 
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $datum >> lanliste-$netz-$datum2.txt
echo "-------------------------------------------------------" >> lanliste-$netz-$datum2.txt
echo Ende: $datum
echo
# Anzeige Ergebnisdatei
less lanliste-$netz-$datum2.txt


IPventur-eng-basic.sh
#! /bin/bash
# IPventur-basic.sh
# need root rights, fping and nmap
# updated: 11.06.2020 MrEnergy64 origin: Linux-User
# Version: 0.7
#
# Start without parameter - exit
clear
if [ -z $1 ]; then
	echo ""
	echo "******************************"
	echo "* IPventur-English Basic 0.7 *"
	echo "******************************"
	echo ""
	echo "NETWORK/CIDR (e.g. 192.168.0.0/24) is needed!"
	echo "(Example: sudo ./IPventur-eng-06.sh 10.0.0.0/23 or 192.168.1.1/32 [Enter])"
	exit
fi
clear
echo "-------------------------------------------------------"
# check if fping and nmap installed
echo "-------------------------------------------------------"
	echo ""
	echo "******************************"
	echo "* IPventur-English Basic 0.7 *"
	echo "******************************"
	echo ""
echo
command -v fping >/dev/null 2>&1 || { echo >&2 "I require fping but it's not installed.  Please install."; exit 1; }
echo Program fping is installed!
echo
command -v nmap >/dev/null 2>&1 || { echo >&2 "I require nmap but it's not installed.  Please install."; exit 1; }
echo Program nmap is installed!
echo
echo
echo "Overview active network systems:"
echo "(to get MAC addresses, start the search network as root/sudo)"
echo "--------------------------------------------------------------"
echo
# aktual date and time will be added to the output file
date2=$(date +%d.%m.%Y-%H:%M:%S)
date3=$(date +%d%m%Y)
echo Start: $date2
echo Command: nmap -A -T5 $1
echo
net2=$(echo $1 | cut -d "/" -f 1)
netO=$net2
# Begin creation output file
echo "-------------------------------------------------------" > lanlist-$netO-$date3.txt
echo "Network items $datum / nmap -A -T5 $1" >> lanlist-$netO-$date3.txt
echo "-------------------------------------------------------"
# Scan network and log to the output file, fping checks online IP's only
for k in $(fping -aq -g $1); do
	echo "scanning...: $k"
	echo "Online: $k" >> lanlist-$netO-$date3.txt
# -R do DNS resolution resolve, -sP check IP and give MAC address with Vendor
	nmap -R -sP $k | awk '/Nmap scan report for/{printf $5" "$6;}/MAC Address:/{print " => "$3" "$4" "$5" "$6;}' | sort >> lanlist-$netO-$date3.txt
# -A Enable OS detection, version detection, script scanning, and tracerout, -T<0-5>: Set timing template (higher is faster) 
	nmap -A -T5 $k | grep -B1 open >> lanlist-$netO-$date3.txt
	echo "---------------------------------------------------" >> lanlist-$netO-$date3.txt
echo >> lanlist-$netO-$date3.txt
done
echo "-------------------------------------------------------"
echo "                     E N D" >> lanlist-$netO-$date3.txt
echo
datum=$(date +%d.%m.%Y-%H:%M:%S)
echo $date2 >> lanlist-$netO-$date3.txt
echo "-------------------------------------------------------" >> lanlist-$netO-$date3.txt
echo End: $date2
echo 
# Display output file
less lanlist-$netO-$date3.txt
